### PR TITLE
add the CBOR tag CDDL (using NLTN) to the testing machinery

### LINKED ISSUE #37: missing `JC<>` definition
We use the `JC` CDDL macro, but we don't define it.

We need to add:

```
JSON-ONLY<J> = J .feature "json"
CBOR-ONLY<C> = C .feature "cbor"

JC<J,C> = JSON-ONLY<J> / CBOR-ONLY<C>
```

maybe in a "conglomerate" CDDL -- (That, or use a CDDL 2 `import` directive.)

### PR DESCRIPTION
**Pull request by thomas-fossati**
Carsten added NLTN support in cddl v0.10.5 :-)

Also add a "collected CDDL" and therefore
Fix #37

### CODE REVIEW

File: draft-ietf-rats-msg-wrap.md [hunk 1]
```diff
@@ -142,10 +142,13 @@ document:
 1. A CMW using a CBOR or JSON array ({{type-n-val}});
 1. A CMW based on CBOR tags ({{cbor-tag}}).
 
+A further CMW "collection" type that holds together multiple CMW conceptual messages is defined in {{cmw-coll}}.
```
Comment (thomas-fossati) [inline draft-ietf-rats-msg-wrap.md:145]:
```suggestion
A further CMW "collection" type that holds together multiple CMW items is defined in {{cmw-coll}}.
```

File: draft-ietf-rats-msg-wrap.md [hunk 2]
```diff
@@ -246,6 +249,30 @@ func CMWTypeSniff(b []byte) (CMW, error) {
...
 ~~~
 
+## CMW Collections {#cmw-coll}
+
+Layered attesters and composite devices ({{Sections 3.2 and 3.3 of -rats-arch}}) generate evidence that consists of multiple parts.
+
+For example, in data center servers, it is not uncommon for separate attesting environments (AE) to serve a subsection of the entire machine.
+One AE might measure and attest to what was booted on the main CPU, while another AE might measure and attest to what was booted on a SmartNIC plugged into a PCIe slot, and a third AE might measure and attest to what was booted on the machine's GPU.
+
+To address the composite Attester use case, this document defines a CMW "collection" as a container that holds several CMW evidence conceptual messages, each with a label that is unique within the scope of the collection.
```
Comment (thomas-fossati) [inline draft-ietf-rats-msg-wrap.md:259]:
```suggestion
To address the composite Attester use case, this document defines a CMW "collection" as a container that holds several CMW items, each with a label that is unique within the scope of the collection.
```

File: draft-ietf-rats-msg-wrap.md [hunk 3]
```diff
@@ -246,6 +249,30 @@ func CMWTypeSniff(b []byte) (CMW, error) {
...
+Layered attesters and composite devices ({{Sections 3.2 and 3.3 of -rats-arch}}) generate evidence that consists of multiple parts.
+
+For example, in data center servers, it is not uncommon for separate attesting environments (AE) to serve a subsection of the entire machine.
+One AE might measure and attest to what was booted on the main CPU, while another AE might measure and attest to what was booted on a SmartNIC plugged into a PCIe slot, and a third AE might measure and attest to what was booted on the machine's GPU.
+
+To address the composite Attester use case, this document defines a CMW "collection" as a container that holds several CMW items, each with a label that is unique within the scope of the collection.
+
+The CMW collection ({{fig-cddl-collection}}) is defined as a CBOR map or JSON object with CMW values.
+The position of a `cmw` entry in the `cmw-collection` is not significant.
+Instead, the labels identify a conceptual message that corresponds to a component of a system.
```
Comment (thomas-fossati) [inline draft-ietf-rats-msg-wrap.md:263]:
```suggestion
Instead, the labels identify a conceptual message that, in the case of a composite attester, should typically correspond to a component of a system.
```

File: draft-ietf-rats-msg-wrap.md [hunk 4]
```diff
@@ -246,6 +249,30 @@ func CMWTypeSniff(b []byte) (CMW, error) {
...
+~~~ cddl
+{::include cddl/cmw-collection.cddl}
+~~~
+{: #fig-cddl-collection artwork-align="left"
+   title="CDDL definition of the CMW collection format"}
+
+Although initially designed for the composite attester use case, the CMW collection can be repurposed for other use cases requiring CMW aggregation.
+
+
+
```
Comment (thomas-fossati) [inline draft-ietf-rats-msg-wrap.md:275]:
```suggestion
```

**Comment by thomas-fossati** [comment]
@henkbirkholz @nedmsmith @hannestschofenig ping!

**Review by hannestschofenig — APPROVED** [review]
It couldn't be better! :-)

**Pull request merged**